### PR TITLE
[PHP 8.1] Document the first class callable syntax.

### PR DESCRIPTION
**Pull request by mumumu**
Added the first class callable syntax docs, based on https://wiki.php.net/rfc/first_class_callable_syntax.

### CODE REVIEW

File: language/functions.xml [hunk 1]
```diff
@@ -1504,6 +1504,127 @@ var_export($x);  // Outputs 1
    </sect2>
   </sect1>
 
+  <sect1 xml:id="functions.first_class_callable_syntax">
+   <title>First class callable syntax</title>
+
+   <para>
+    The first class callable syntax is introduced as of PHP 8.1.0, as a way to creating <link linkend="functions.anonymous">anonymous functions</link> from <link linkend="language.types.callable">callable</link>.
+    It supersedes existing callable encodings using strings and arrays.
```
Comment (kamil-tekiela) [inline language/functions.xml:1512]: What are "callable encodings"? Is this phrase defined somewhere? I can only find this phrase used once in RFC and it is a little bit ambiguous. Maybe we should explain what it means in the manual.

File: language/functions.xml [hunk 2]
```diff
@@ -1504,6 +1504,127 @@ var_export($x);  // Outputs 1
...
+    The advantage of this syntax is that it is accessible to static analysis, and respects the scope at the point where the callable is created.
```
Comment (kamil-tekiela) [inline language/functions.xml:1513]:
```suggestion
    The advantage of this syntax is that it is accessible to static analysis, and uses the scope at the point where the callable is acquired.
```

File: language/types/callable.xml [hunk 3]
```diff
@@ -48,6 +48,12 @@
    passed to a callback parameter.
   </para>
 
+  <note>
+   <para>
+    As of PHP 8.1.0, anonymous functions can also be created by using the <link linkend="functions.first_class_callable_syntax">first class callable syntax</link>.
```
Comment (kamil-tekiela) [inline language/types/callable.xml:53]:
```suggestion
    As of PHP 8.1.0, anonymous functions can also be created using the <link linkend="functions.first_class_callable_syntax">first class callable syntax</link>.
```

File: language/functions.xml (hunk 1 above)
Comment (mumumu) [reply language/functions.xml]: > What are "callable encodings"? 

It means traditional callable syntax as following, so I replaced `encodings` with `syntax`.

```php
// traditional callable using string, array
$f7 = 'strlen'(...);
$f8 = [$obj, 'method'](...);
```

**Pull request merged**